src/lp.rs

```diff
@@ -0,0 +1,428 @@
+    "\\bhttps://open.spotify.com/album/([a-zA-Z0-9]+)(?:\\?[a-zA-Z?=&]*)\\b";
+
+const SPOTIFY_PLAYLIST_RE: &str =
+    "\\bhttps://open.spotify.com/playlist/([a-zA-Z0-9]+)(?:\\?[a-zA-Z?=&]*)\\b";
+
+#[derive(Command, Debug)]
+#[cmd(name = "lp", desc = "Check if listening party is going")]
+pub struct CurrentLP {}
+
+#[async_trait]
+impl BotCommand for CurrentLP {
+    type Data = Handler;
+    async fn run(
+        self,
+        data: &Handler,
+        _ctx: &Context,
+        interaction: &CommandInteraction,
+    ) -> anyhow::Result<CommandResponse> {
+        let channel = interaction.channel_id;
+        let lpmod = data.module::<LP>().unwrap();
+        let lps = lpmod.last_pinged.read().await;
+        let lp = lps.get(&channel);
+        let response = match lp {
+            None => CommandResponse::Public(
+                "There is no listening party at the moment.".to_string(),
+            ),
+            Some(lpinfo) => CommandResponse::Embed(lpinfo.build_embed()),
+        };
+        Ok(response)
+    }
+}
+
+pub type PingedMap = Arc<RwLock<HashMap<ChannelId, LPInfo>>>;
+
+pub struct LP {
+    last_pinged: PingedMap,
+}
+
+impl Clone for LP {
+    fn clone(&self) -> Self {
+        LP {
+            last_pinged: self.last_pinged.clone(),
+        }
+    }
+}
+
+#[async_trait]
+impl polls::ModPollReadyHandler for LP {
+    async fn ready(&self, channelid: &ChannelId) {
+        self.start_lp(channelid).await;
+    }
+}
+
+// Roles used for pinging listening parties
+const LP_ROLES: &'static [u64] = &[
+    1198354637137391709, // `@Listening Party`` in test guild
+                         // TODO: Make this configurable? Fetch via name?
+];
+
+impl LP {
+    pub fn new() -> Self {
+        LP {
+            last_pinged: Default::default(),
+        }
+    }
+
+    // Handle messages to remember the last pinged album
+    pub async fn handle_message<C: BaseClient>(
+        &self,
+        client: &C,
+        msg: &Message,
+    ) {
+        let msg_txt: &str = &msg.content;
+
+        // Check if the specified roles were mentioned
+        if msg
+            .mention_roles
+            .iter()
+            .any(|&role| LP_ROLES.iter().contains(&role.get()))
+        {
+            let mb_pl = 'regex: {
+                // Check for spotify playlist URL
+                if let Some(caps) =
+                    Regex::new(&SPOTIFY_ALBUM_RE).unwrap().captures(&msg_txt)
+                {
+                    break 'regex (LPInfo::from_spotify_album_id(
+                        client, &caps[1],
+                    )
+                    .await);
+                }
+                // Check for spotify playlist URL
+                if let Some(caps) =
+                    Regex::new(&SPOTIFY_PLAYLIST_RE).unwrap().captures(&msg_txt)
+                {
+                    break 'regex LPInfo::from_spotify_playlist_id(
+                        client, &caps[1],
+                    )
+                    .await;
+                }
```

Comment:
This could be moved to a separate function to avoid the weird break syntax (cool feature though, didn't know about it).

Reply:
I moved it to `LPInfo::from_match_string`